### PR TITLE
Support for forward/back mousebuttons on WX backend

### PR DESCRIPTION
**Pull request by Ptrskay3**
## PR Summary
This PR adds the support for using the `MouseButton.FORWARD` and `MouseButton.BACKWARD` on the WX backend.

## PR Checklist

- [ ] Has pytest style unit tests (and `pytest` passes).
- [x] Is [Flake 8](https://flake8.pycqa.org/en/latest/) compliant (run `flake8` on changed files to check).
- [ ] New features are documented, with examples if plot related.
- [N/A] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).
- [x] Conforms to Matplotlib style conventions (install `flake8-docstrings` and run `flake8 --docstring-convention=all`).
- [ ] New features have an entry in `doc/users/next_whats_new/` (follow instructions in README.rst there).
- [ ] API changes documented in `doc/api/next_api_changes/` (follow instructions in README.rst there).

### CODE REVIEW

**Comment by tacaswell** [comment]
This seems reasonable to me.  

Do we handle this in the other GUI toolkits already?  If not would you be willing to add it to them as well (or atleast make issues to add them?)?

**Comment by Ptrskay3** [comment]
In Qt it's [already it there](https://github.com/matplotlib/matplotlib/blob/master/lib/matplotlib/backends/backend_qt.py#L209). GTK3 [also supports it](https://github.com/matplotlib/matplotlib/pull/12904#issuecomment-466625549), although I haven't verified that. AFAICT in Tk there is no built in way to detect the forward/back buttons. I haven't looked at other backends yet.

File: lib/matplotlib/backends/backend_wx.py [hunk 1]
```diff
@@ -766,6 +772,8 @@ def _onMouseButton(self, event):
             wx.MOUSE_BTN_LEFT: MouseButton.LEFT,
             wx.MOUSE_BTN_MIDDLE: MouseButton.MIDDLE,
             wx.MOUSE_BTN_RIGHT: MouseButton.RIGHT,
+            wx.MOUSE_BTN_AUX1: MouseButton.FORWARD,
+            wx.MOUSE_BTN_AUX2: MouseButton.BACK,
```
Comment (timhoffm) [inline lib/matplotlib/backends/backend_wx.py:776]:
```suggestion
            wx.MOUSE_BTN_AUX1: MouseButton.BACK,
            wx.MOUSE_BTN_AUX2: MouseButton.FORWARD,
```
When testing, the assignment seems to be wrong. Please recheck.

Comment (Ptrskay3) [reply lib/matplotlib/backends/backend_wx.py]: Yes, I flipped that one. I always have a hard time deciding which is forward and which one is back.

**Review by timhoffm — APPROVED** [review]

**Review by tacaswell — APPROVED** [review]

**Comment by timhoffm** [comment]
Thanks @Ptrskay3!

**Pull request merged**